docs/flexible-sync.js

```diff
@@ -136,9 +136,12 @@ var SubscriptionsState = {
  * Class representing the common functionality for the {@link Realm.App.Sync.SubscriptionSet} and
  * {@link Realm.App.Sync.SubscriptionSet} classes
  *
- * The {@link Realm.App.Sync.Subscription}s in a SubscriptionSet can be accessed as an array, e.g.
- * `realm.subscriptions[0]`. This array is readonly – SubscriptionSets can only be
- * modified inside a {@link Realm.App.Sync.SubscriptionSet.update} callback.
+ * SubscriptionSets can only be modified inside a {@link SubscriptionSet.update} callback.
+ *
+ * The SubscriptionSet is an iterable; thus, if absolutely needed, the contained
+ * {@link Subscription}s can be accessed in `for-of` loops or spread into an `Array`
+ * for access to the ECMAScript Array API, e.g. `[...realm.subscriptions][0]`. Directly
+ * accessing the SubscriptionSet as an array is deprecated.
```

Comment:
```suggestion
 * The SubscriptionSet is an iterable; thus, the contained
 * {@link Subscription}s can be accessed in `for-of` loops or spread into an `Array`
 * for access to the ECMAScript Array API, e.g. `[...realm.subscriptions][0]`. Directly
 * accessing the SubscriptionSet as if it was an array is deprecated.
```

I don't think we need to moralize against for-of iteration here. Is there any reason to discourage it?